support-workers/src/main/scala/com/gu/zuora/subscriptionBuilders/SubscribeItemBuilder.scala

```diff
@@ -1,53 +1,109 @@
-    }
-    ContactDetails(
-      firstName = giftRecipient.fold(user.firstName)(_.firstName),
-      lastName = giftRecipient.fold(user.lastName)(_.lastName),
-      workEmail = email,
-      address1 = address.lineOne,
-      address2 = address.lineTwo,
-      city = address.city,
-      postalCode = address.postCode,
-      country = address.country,
-      state = address.state,
-      deliveryInstructions = deliveryInstructions
+  private def buildAccount(salesForceContact: SalesforceContactRecord, maybePaymentMethod: Option[PaymentMethod]) = {
+    Account(
+      name = salesForceContact.AccountId, //We store the Salesforce Account id in the name field
+      currency = currency,
+      crmId = salesForceContact.AccountId, //Somewhere else we store the Salesforce Account id
+      sfContactId__c = salesForceContact.Id,
+      identityId__c = user.id,
+      paymentGateway = maybePaymentMethod.map(_.paymentGateway),
+      createdRequestId__c = requestId.toString,
+      autoPay = maybePaymentMethod.isDefined
     )
   }
 
-  private def buildAccount(state: CreateZuoraSubscriptionState) = Account(
-    name = state.salesforceContacts.recipient.AccountId, //We store the Salesforce Account id in the name field
-    currency = state.product.currency,
-    crmId = state.salesforceContacts.recipient.AccountId, //Somewhere else we store the Salesforce Account id
-    sfContactId__c = state.salesforceContacts.recipient.Id,
-    identityId__c = state.user.id,
-    paymentGateway = state.paymentMethod.left.toOption.map(_.paymentGateway),
-    createdRequestId__c = state.requestId.toString,
-    autoPay = state.paymentMethod.isLeft
+  def buildProductSubscription(
+    productRatePlanId: ProductRatePlanId,
+    ratePlanCharges: List[RatePlanChargeData] = Nil,
+    contractEffectiveDate: LocalDate = LocalDate.now(DateTimeZone.UTC),
+    contractAcceptanceDate: LocalDate = LocalDate.now(DateTimeZone.UTC),
+    readerType: ReaderType,
+    autoRenew: Boolean = true,
+    initialTerm: Int = 12,
+    initialTermPeriodType: PeriodType = Month,
+    redemptionCode: Option[RawRedemptionCode] = None,
+    giftNotificationEmailDate: Option[LocalDate] = None,
+  ): SubscriptionData =
+    SubscriptionData(
+      List(
+        RatePlanData(
+          RatePlan(productRatePlanId),
+          ratePlanCharges,
+          Nil
+        )
+      ),
+      Subscription(
+        contractEffectiveDate = contractEffectiveDate,
+        contractAcceptanceDate = contractAcceptanceDate,
+        termStartDate = contractEffectiveDate,
+        createdRequestId = requestId.toString,
+        readerType = readerType,
+        autoRenew = autoRenew,
+        initialTerm = initialTerm,
+        initialTermPeriodType = initialTermPeriodType,
+        redemptionCode = redemptionCode,
+        giftNotificationEmailDate = giftNotificationEmailDate,
+      )
+    )
+
+}
+
+object SubscribeItemBuilder {
+
+  def buildContactDetails(
```

Comment:
this is now just a smart constructor for `ContactDetails` so we could call this `ContactDetails.apply`, the onyl issue there was scala can't accept them both having default arguments.  WE could call that another name e.g. `ContactDetails.create`, but I have stuck it here for now.